src/middlewared/middlewared/plugins/boot_/environments.py

```diff
@@ -142,14 +142,14 @@ def activate(self, data):
             )
         else:
             run_zectl_cmd(["activate", data["id"]])
-            return self.query([["id", "=", data["id"]]])
+            return self.query([["id", "=", data["id"]]], {"get": True})
```

Comment:
You can just use `get_instance`

Reply:
it's actually `get_instance__sync` but yeah, I'll change it. Thanks

Reply:
hmm, nope, I don't have access to that because of inherited class. I'll leave it as-is for now.